### DISCUSSION TITLE
SAHI not compatible with Shapely==1.8.5

### DISCUSSION THREAD

**Discussion by DamianBisewski** [obss/sahi#1045] [Q&A] [answered]
Hello, I would like to report an issue with the compatibility of the SAHI package with Shapely version 1.8.5.
---------------------------------------------------------------------------
In this configuration, when attempting to use sahi.utils.coco, the following error occurs:
from sahi import AutoDetectionModel
from sahi.predict import get_sliced_prediction, get_prediction
from [sahi.utils.cv](http://sahi.utils.cv/) import read_image_as_pil
ImportError                               Traceback (most recent call last)
Cell In[27], line 2
      1 from shapely.geometry import Polygon
----> 2 from sahi import AutoDetectionModel
      3 from sahi.predict import get_sliced_prediction, get_prediction
      4 from sahi.utils.cv import read_image_as_pil

File /opt/conda/lib/python3.10/site-packages/sahi/__init__.py:3
      1 __version__ = "0.11.16"
----> 3 from sahi.annotation import BoundingBox, Category, Mask
      4 from sahi.auto_model import AutoDetectionModel
      5 from sahi.models.base import DetectionModel

File /opt/conda/lib/python3.10/site-packages/sahi/annotation.py:9
      5 from typing import Dict, List, Optional
      7 import numpy as np
----> 9 from sahi.utils.coco import CocoAnnotation, CocoPrediction
     10 from sahi.utils.cv import (
     11     get_bbox_from_bool_mask,
     12     get_bool_mask_from_coco_segmentation,
     13     get_coco_segmentation_from_bool_mask,
     14 )
     15 from sahi.utils.shapely import ShapelyAnnotation

File /opt/conda/lib/python3.10/site-packages/sahi/utils/coco.py:17
     14 from typing import Dict, List, Optional, Set, Union
     16 import numpy as np
---> 17 from shapely import MultiPolygon
     18 from shapely.validation import make_valid
     19 from tqdm import tqdm

ImportError: cannot import name 'MultiPolygon' from 'shapely' (/opt/conda/lib/python3.10/site-packages/shapely/__init__.py)

The fix is trivial, just replace line 17 in sahi/utils/coco.py:
from shapely import MultiPolygon
with:
from shapely.geometry import MultiPolygon
This change will make SAHI compatible with Shapely 1.8.5 (and newer versions as well).

**Comment by fcakyon** [comment, marked as answer]
@DamianBisewski, this issue is fixed in the latest pr thanks to @Alias-z 🚀 
https://github.com/obss/sahi/pull/1047
